pages/index.js

```diff
@@ -23,6 +25,32 @@ export default function Home({ topicsData, errorCode }) {
       </div>
     );
 
+  React.useEffect(() => {
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition(getPermissionState);
+      if (permission === "granted") {
+        getGeo();
+      }
+    } else {
+      //will add functionality for older browsers and those who don't give geolocation permission to select a province at a later date
+      alert("Sorry, browser does not support geolocation!");
+    }
+  });
+
+  async function getGeo() {
+    const res = await fetch("http://ip-api.com/json");
+    const data = await res.json();
+    setRegion(data.region);
+  }
+
+  function getPermissionState() {
+    navigator.permissions
+      .query({ name: "geolocation" })
+      .then(function (permissionStatus) {
+        setPermission(permissionStatus.state);
```

Comment:
I got the permission prompt from firefox which I granted (without checkbox for remember my answer, so that it will ask each time)  the issue is the permissionStatus.state === prompt. 
```
permission status  
PermissionStatus { state: "prompt", onchange: null }
```
so the geoCode is not executed

Reply:
My fault for not testing it on Firefox. Will have to rethink the logic on this, though it's seemingly the only browser having issues